calabash/Classes/FranklyServer/LPRouter.m

```diff
@@ -206,11 +206,21 @@ - (BOOL) supportsMethod:(NSString *) method atPath:(NSString *) path {
   if ([[NSThread currentThread] isMainThread]) {
     return [self httpResponseOnMainThreadForMethod:method URI:path];
   } else {
-    __weak typeof(self) wself = self;
-    __block NSObject<LPHTTPResponse> *result = nil;
-    dispatch_sync(dispatch_get_main_queue(), ^{
-      result = [wself httpResponseOnMainThreadForMethod:method URI:path];
-    });
+    // Use performSelectorOnMainThread instead of dispatch_sync because using a dispatch method prevents
```

Comment:
Yes, if I add a test the comment can be removed. This isn't something I have time to do this week though.